docs/items/tools.md

```diff
@@ -0,0 +1,246 @@
+            List.of(Tiers.STONE),
+            //A list of tiers that are considered higher than the type being added. For example, iron is higher than copper.
+            //We don't need to add diamond and netherite here because those are already higher than iron.
+            List.of(Tiers.IRON)
+    );
+}
+```
+
+If you want to check if a tier is applicable for a block state, you can call `TierSortingRegistry#isCorrectTierForDrops`.
+
+## Custom Tool Types
+
+Custom tool types can be created by extending `DiggerItem` (or `TieredItem` if you are making custom weapon types). They don't need too big of a setup, it is an item class like any other, with all implications that has.
+
+One thing worth noting is the parameters of `DiggerItem`. The first four parameters are the same as for its subclasses (see the explanation for `SwordItem` above), while the fifth parameter is the `mineable` tag for the tool type. Generally, the format here is `<mod_id>:mineable/<tool_type>`, though `forge` can be used as the namespace too if you expect other mods to add similar tools. For example, [Farmer's Delight][farmersdelight] uses a `forge:mineable/knives` tag.
+
+## `ToolAction`s
+
+`ToolAction`s are an abstraction over what a tool can and cannot do. This includes both left-click and right-click behavior. NeoForge provides default `ToolAction`s in the `ToolActions` class:
+
+- Digging actions. These exist for all four `DiggerItem` types as mentioned above, as well as sword and shears digging.
+- Axe right-click actions for stripping (logs), scraping (oxidized copper) and unwaxing (waxed copper).
+- Shear actions for harvesting (honeycombs), carving (pumpkins) and disarming (tripwires).
+- Actions for shovel flattening (dirt paths), sword sweeping, hoe tilling, shield blocking, and fishing rod casting.
+
+To create your own `ToolAction`s, use `ToolAction#get` - it will create a new `ToolAction` if needed. Then, in a custom tool type, override `IItemExtension#canPerformAction` as needed.
+
+To query if an `ItemStack` can perform a certain `ToolAction`, call `IItemStackExtension#canPerformAction`. Note that this works on any `Item`, not just tools.
+
+## Armor
+
+Similar to tools, armor uses a tier system (although a different one). What is called `Tier` for tools is called `ArmorMaterial` for armors. Like above, this example shows how to add copper armor; this can be adapted as needed. For the vanilla values, see the `ArmorMaterials` enum.
+
+```java
+// We place copper somewhere between chainmail and iron.
+public static final ArmorMaterial COPPER_ARMOR_MATERIAL = new ArmorMaterial() {
+    // The name of the armor material. Mainly determines where the armor texture is.
+    @Override
+    public String getName() {
+        return "copper";
+    }
+
+    // Override for StringRepresentable. Should generally return the same as getName().
+    @Override
+    public String getSerializedName() {
+        return "copper";
```

Comment:
Should have modid prefix to guarantee uniqueness. Better to have this delegate to `#getName`.